src/server/operators/select.rs

```diff
@@ -115,34 +118,40 @@ impl Conditions {
     fn eval(&self, shared_block: &Arc<RwLock<Block>>) -> ResultSet {
         match self {
             // If a Leaf, return results.
-            Conditions::Leaf(cond) => cond.eval(shared_block),
+            Conditions::Leaf(cond) => {
+                let mut r = cond.eval(shared_block);
+                r.unpack(shared_block);
+                r
+            }
             // If an And, intersect the results.
             Conditions::And(b1, b2) => {
-                let r1 = (*b1).eval(shared_block);
+                let mut r1 = (*b1).eval(shared_block);
                 let r2 = (*b2).eval(shared_block);
-                r1.intersection(r2)
+                r1.intersection(r2, shared_block);
+                r1.unpack(shared_block);
+                r1
             }
             // If an or, union the results.
             Conditions::Or(b1, b2) => {
-                let r1 = (*b1).eval(shared_block);
+                let mut r1 = (*b1).eval(shared_block);
                 let r2 = (*b2).eval(shared_block);
-                r1.union(r2)
+                r1.union(r2, shared_block);
+                r1.unpack(shared_block);
+                r1
```

Comment:
Union returns an unpacked, so this unpack shouldn't do anything right?

Reply:
Ah not necessarily: if a union encounters two packed ResultSets, AND both ResultSets don't have filters, then it doesn't unpack them.